lib/events.js

```diff
@@ -664,6 +641,38 @@ function createIterResult(value, done) {
   return { value, done };
 }
 
+function addErrorHandlerIfEventEmitter(emitter, handler, flags) {
+  if (typeof emitter.on === 'function') {
+    eventTargetAgnosticAddListener(emitter, 'error', handler, flags);
+  }
+}
+
+function eventTargetAgnosticRemoveListener(emitter, name, listener, flags) {
+  if (typeof emitter.removeEventListener === 'function') {
+    emitter.removeEventListener(name, listener, flags);
+  } else if (typeof emitter.removeListener === 'function') {
+    emitter.removeListener(name, listener);
+  } else {
+    throw new ERR_INVALID_ARG_TYPE('emitter', 'EventEmitter', emitter);
+  }
+}
+
+function eventTargetAgnosticAddListener(emitter, name, listener, flags) {
+  if (typeof emitter.addEventListener === 'function') {
+    // EventTarget does not have `error` event semantics like Node
+    // EventEmitters, we do not listen to `error` events here.
+    emitter.addEventListener(name, (arg) => { listener(arg); }, flags);
```

Comment:
Why is the additional wrapping of `listener` needed? (since `dispatchEvent` will call it with 1 argument anyway IIUC)

Reply:
Basically just to ensure only one argument is passed. Though our EventTarget (and any other implementation I know) only pass a single argument. I don't feel strongly about this and if you do I'll gladly change it to not wrap.

Reply:
Not really, just noticed wrapping and wanted to make sure it was intentional.